pytorch_lightning/loops/closure.py

```diff
@@ -53,25 +54,29 @@ def __init__(self) -> None:
         super().__init__()
         self._result: Optional[ClosureResult] = None
 
-    def get_result(self) -> Optional[ClosureResult]:
```

Comment:
the naming here was intentional. We are only allowed to access the result once. In that sense, it really is not a property. I vote for keeping this a method.

Reply:
Sounds good, changed `get_result` to `consume_result` to make it more explicit